### PR TITLE
docs: Adding Interactivity Overview translations

### LINKED ISSUE #452: Adding Interactivity
Page: https://id.react.dev/learn/adding-interactivity

Assignee: @chrsep

### PR DESCRIPTION
**Pull request by chrsep**
Added translations for `Adding Interactivity: Overview` page.

Closes #452

### CODE REVIEW

File: src/content/learn/adding-interactivity.md [hunk 1]
```diff
@@ -1,38 +1,38 @@
 ---
-title: Adding Interactivity
+title: Menambahkan Interaktivitas
 ---
 
 <Intro>
 
-Some things on the screen update in response to user input. For example, clicking an image gallery switches the active image. In React, data that changes over time is called *state.* You can add state to any component, and update it as needed. In this chapter, you'll learn how to write components that handle interactions, update their state, and display different output over time.
+Beberapa hal di layar berubah mengikuti input dari pengguna. Contohnya, mengeklik sebuah galeri gambar bisa mengganti gambar yang sedang aktif. Di React, data yang berubah seiring waktu disebut *state.* Anda dapat menambahkan state ke komponen apapun, dan mengubahnya sesuai kebutuhan. Di bab ini, Anda akan belajar cara menulis komponen yang dapat menangani interaksi, mengubah state yang dimilikinya, dan menampilkan output yang berbeda seiring berjalannya waktu.
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:7]: Input -> Masukan
Output -> Keluaran 
```suggestion
Beberapa hal di layar berubah mengikuti masukan dari pengguna. Contohnya, mengeklik sebuah galeri gambar bisa mengganti gambar yang sedang aktif. Di React, data yang berubah seiring waktu disebut *state.* Anda dapat menambahkan state ke komponen apapun, dan mengubahnya sesuai kebutuhan. Di bab ini, Anda akan belajar cara menulis komponen yang dapat menangani interaksi, mengubah state yang dimilikinya, dan menampilkan keluaran yang berbeda seiring berjalannya waktu.
```

File: src/content/learn/adding-interactivity.md [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
-* [How React updates the UI in two phases](/learn/render-and-commit)
-* [Why state doesn't update right after you change it](/learn/state-as-a-snapshot)
-* [How to queue multiple state updates](/learn/queueing-a-series-of-state-updates)
-* [How to update an object in state](/learn/updating-objects-in-state)
-* [How to update an array in state](/learn/updating-arrays-in-state)
+* [Cara menangani events yang di-mulai oleh pengguna](/learn/responding-to-events)
+* [Cara membuat komponen mengingat informasi dengan menggunakan state](/learn/state-a-components-memory)
+* [Cara React memperbaharui UI dalam dua fase](/learn/render-and-commit)
+* [Mengapa state tidak langsung terbaharui setelah Anda mengubahnya](/learn/state-as-a-snapshot)
+* [Cara meng-queue beberapa perubahan state](/learn/queueing-a-series-of-state-updates)
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:17]:
```suggestion
* [Cara mengantrikan beberapa perubahan state](/learn/queueing-a-series-of-state-updates)
```

File: src/content/learn/adding-interactivity.md [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
+* [Cara mengubah object didalam state](/learn/updating-objects-in-state)
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:18]:
```suggestion
* [Cara mengubah object di dalam state](/learn/updating-objects-in-state)
```

File: src/content/learn/adding-interactivity.md [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
+* [Cara mengubah array didalam state](/learn/updating-arrays-in-state)
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:19]:
```suggestion
* [Cara mengubah array di dalam state](/learn/updating-arrays-in-state)
```

File: src/content/learn/adding-interactivity.md [hunk 5]
```diff
@@ -1,38 +1,38 @@
...
 </YouWillLearn>
 
-## Responding to events {/*responding-to-events*/}
+## Menanggapi events {/*responding-to-events*/}
 
-React lets you add *event handlers* to your JSX. Event handlers are your own functions that will be triggered in response to user interactions like clicking, hovering, focusing on form inputs, and so on.
+React memungkinkan Anda untuk menambakan *event handlers* ke JSX Anda. *Event handlers* adalah fungsi milik Anda yang akan dipanggil sebagai respon terhadap interaksi dari pengguna seperti klik, *hover*, fokus pada input form, dan lain-lain.
 
-Built-in components like `<button>` only support built-in browser events like `onClick`. However, you can also create your own components, and give their event handler props any application-specific names that you like.
+Komponen bawaan seperti `<button>` hanya mendukung *event* bawaan dari browser seperti `onClick`. Namun, Anda juga dapat membuat komponen Anda sendiri, dan memberikannya *prop event handler* dengan nama apapun, spesifik terhadap aplikasi Anda.
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:27]:
```suggestion
Komponen bawaan seperti `<button>` hanya mendukung *event* bawaan dari peramban seperti `onClick`. Namun, Anda juga dapat membuat komponen Anda sendiri, dan memberikannya *prop event handler* dengan nama apapun, spesifik terhadap aplikasi Anda.
```

File: src/content/learn/adding-interactivity.md [hunk 6]
```diff
@@ -314,13 +314,13 @@ label, textarea { margin-bottom: 10px; display: block; }
 
 <LearnMore path="/learn/state-as-a-snapshot">
 
-Read **[State as a Snapshot](/learn/state-as-a-snapshot)** to learn why state appears "fixed" and unchanging inside the event handlers.
+Baca **[State sebagai Snapshot](/learn/state-as-a-snapshot)** untuk mempelajari mengapa state terlihat "tetap" dan tak berubah di dalam *event handler*.
 
 </LearnMore>
 
-## Queueing a series of state updates {/*queueing-a-series-of-state-updates*/}
+## Meng-queue serangkaian perubahan state {/*queueing-a-series-of-state-updates*/}
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:321]:
```suggestion
## Mengantrikan serangkaian perubahan state {/*queueing-a-series-of-state-updates*/}
```

File: src/content/learn/adding-interactivity.md [hunk 7]
```diff
@@ -366,7 +366,7 @@ setScore(score + 1); // setScore(0 + 1);
 console.log(score);  // 0
 ```
 
-You can fix this by passing an *updater function* when setting state. Notice how replacing `setScore(score + 1)` with `setScore(s => s + 1)` fixes the "+3" button. This lets you queue multiple state updates.
+Anda bisa memperbaiki ini dengan memberikan *updater function* ketika mengubah state. Perhatikan bagaimana mengganti `setScore(score + 1)` dengan `setScore(s => s + 1)` memperbaiki tombol "+3". Cara ini memungkinkan Anda untuk meng-*queue* beberapa perubahan state.
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:369]:
```suggestion
Anda bisa memperbaiki ini dengan memberikan *updater function* ketika mengubah state. Perhatikan bagaimana mengganti `setScore(score + 1)` dengan `setScore(s => s + 1)` memperbaiki tombol "+3". Cara ini memungkinkan Anda untuk mengantrikan beberapa perubahan state.
```

File: src/content/learn/adding-interactivity.md [hunk 8]
```diff
@@ -402,15 +402,16 @@ button { display: inline-block; margin: 10px; font-size: 20px; }
 
 <LearnMore path="/learn/queueing-a-series-of-state-updates">
 
-Read **[Queueing a Series of State Updates](/learn/queueing-a-series-of-state-updates)** to learn how to queue a sequence of state updates.
+Baca **[Meng-queue serangkaian perubahan state](/learn/queueing-a-series-of-state-updates)** untuk mempelajari cara meng-*queue* sebuah rentetan perubahan state.
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:405]:
```suggestion
Baca **[Mengantrikan serangkaian perubahan state](/learn/queueing-a-series-of-state-updates)** untuk mempelajari cara meng-*queue* sebuah rentetan perubahan state.
```

File: src/content/learn/adding-interactivity.md [hunk 9]
```diff
@@ -633,13 +634,13 @@ img { width: 200px; height: 200px; }
 
 <LearnMore path="/learn/updating-objects-in-state">
 
-Read **[Updating Objects in State](/learn/updating-objects-in-state)** to learn how to update objects correctly.
+Baca **[Mengubah Object didalam State](/learn/updating-objects-in-state)** untuk mempelajari cara mengubah *object* dengan benar.
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:637]:
```suggestion
Baca **[Mengubah Object di dalam State](/learn/updating-objects-in-state)** untuk mempelajari cara mengubah *object* dengan benar.
```

File: src/content/learn/adding-interactivity.md [hunk 10]
```diff
@@ -633,13 +634,13 @@ img { width: 200px; height: 200px; }
...
 
 </LearnMore>
 
-## Updating arrays in state {/*updating-arrays-in-state*/}
+## Mengubah array didalam state {/*updating-arrays-in-state*/}
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:641]:
```suggestion
## Mengubah array di dalam state {/*updating-arrays-in-state*/}
```

File: src/content/learn/adding-interactivity.md [hunk 11]
```diff
@@ -791,12 +792,12 @@ function ItemList({ artworks, onToggle }) {
 
 <LearnMore path="/learn/updating-arrays-in-state">
 
-Read **[Updating Arrays in State](/learn/updating-arrays-in-state)** to learn how to update arrays correctly.
+Baca **[Mengubah Array didalam State](/learn/updating-arrays-in-state)** untuk mempelajari cara mengubah *array* dengan benar.
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:795]:
```suggestion
Baca **[Mengubah Array di dalam State](/learn/updating-arrays-in-state)** untuk mempelajari cara mengubah *array* dengan benar.
```

**Comment by r17x** [comment]
any updates, @chrsep?

**Comment by resir014** [comment]
Hi @chrsep, are you still available to work on this PR? If you're unable to finish within the next day, are you okay with me taking over this PR?

**Comment by chrsep** [comment]
Hi @r17x @resir014 , just saw that the PR got reviewed last midnight, I just got back online. I've updated the PR.

**Review by r17x — APPROVED** [review]
I'm approved but need confirm from @chrsep  for 1 unresolved review by @mazipan

**Comment by chrsep** [comment]
> I'm approved but need confirm from @chrsep for 1 unresolved review by @mazipan

oops, missed that one. Updated.

File: src/content/learn/adding-interactivity.md [hunk 12]
```diff
@@ -229,43 +229,43 @@ button {
...
+Sebelum komponen Anda ditampilkan di layar, mereka harus di *render* oleh React. Memahami langkah-langkah dalam proses ini akan membantu Anda berpikir tentang bagaimana kode Anda akan dijalankan dan menjelaskan perilakunya.
 
-Imagine that your components are cooks in the kitchen, assembling tasty dishes from ingredients. In this scenario, React is the waiter who puts in requests from customers and brings them their orders. This process of requesting and serving UI has three steps:
+Bayangkan bahwa komponen Anda adalah koki di dapur, menyusun hidangan lezat dari bahan-bahan dasar. Dalam skenario ini, React adalah pelayan yang mengajukan permintaan dari pelanggan dan membawakan pesanan mereka. Proses permintaan dan pelayanan UI ini memiliki tiga langkah:
 
-1. **Triggering** a render (delivering the diner's order to the kitchen)
-2. **Rendering** the component (preparing the order in the kitchen)
-3. **Committing** to the DOM (placing the order on the table)
+1. **Memicu** *render* (mengirimkan pesanan pelanggan ke dapur)
+2. **Merender** komponen (menyiapkan pesanan di dapur)
```
Comment (r17x) [inline src/content/learn/adding-interactivity.md:243]: Once more. We have been updated the terms and for word like "merender" need to changes "me-*render*" except in **Heading** ("me-render").

**Review by r17x — CHANGES_REQUESTED** [review]

File: src/content/learn/adding-interactivity.md [hunk 13]
```diff
@@ -229,43 +229,43 @@ button {
...
-Read **[Render and Commit](/learn/render-and-commit)** to learn the lifecycle of a UI update.
+Baca **[Render dan Commit](/learn/render-and-commit)** untuk mempelajari *lifecycle* dari perubahan UI.
 
 </LearnMore>
 
-## State as a snapshot {/*state-as-a-snapshot*/}
+## State sebagai snapshot {/*state-as-a-snapshot*/}
 
-Unlike regular JavaScript variables, React state behaves more like a snapshot. Setting it does not change the state variable you already have, but instead triggers a re-render. This can be surprising at first!
+Tidak seperti variabel Javascript biasa, state di React berperilaku lebih seperti *snapshot*. Mengubah state tidaklah mengubah variabel state yang Anda miliki sekarang, tetapi akan memicu re-render. Ini bisa mengejutkan pada awalnya!
```
Comment (mazipan) [inline src/content/learn/adding-interactivity.md:260]: Semua re-render atau rerender = *render* ulang.
Dengan italic di render.

```suggestion
Tidak seperti variabel Javascript biasa, state di React berperilaku lebih seperti *snapshot*. Mengubah state tidaklah mengubah variabel state yang Anda miliki sekarang, tetapi akan memicu *render* ulang. Ini bisa mengejutkan pada awalnya!
```

File: src/content/learn/adding-interactivity.md (hunk 12 above)
Comment (chrsep) [reply src/content/learn/adding-interactivity.md]: Thanks, updated.

File: src/content/learn/adding-interactivity.md (hunk 13 above)
Comment (chrsep) [reply src/content/learn/adding-interactivity.md]: Thanks, updated.

**Review by r17x — APPROVED** [review]

**Pull request merged**